integration_test/warehouse/testdata/docker-compose.ssh-server.yml

```diff
@@ -0,0 +1,33 @@
+version: "3.9"
+
+services:
+  ssh-server:
+    image: lscr.io/linuxserver/openssh-server:latest
+    environment:
+      - PUBLIC_KEY_FILE=/test_key.pub
+      - USER_NAME=rudderstack #optional
+      - DOCKER_MODS=linuxserver/mods:openssh-server-ssh-tunnel # https://github.com/linuxserver/docker-mods/tree/openssh-server-ssh-tunnel
+    ports:
+      - "2222"
+    volumes:
+      - type: bind
+        source: ./ssh-server/test_key.pub
+        target: /test_key.pub
+        read_only: true
+    healthcheck:
+      test: [ "CMD", "grep", "Server listening on :: port 2222", "/config/logs/openssh/current" ]
+      interval: 1s
+      timeout: 1s
+      retries: 60
+    depends_on:
+      - db-private-postgres
+  db-private-postgres:
+    image: postgres:14.1-alpine
```

Comment:
Good point. Thanks, @fracasula for bringing this up. There are other few places as well where we are not using the latest. I will update them in a separate PR.

Reply:
https://github.com/rudderlabs/rudder-server/pull/4092

Reply:
The only concern here is the backward compatibility since we will be using older version [server(PostgreSQL 11.21, PostgreSQL 12.9) and warehouse (PostgreSQL 12.9, PostgreSQL 11.6)] in Production, as compared to [Postgres latest (PostgreSQL 16)] in tests.

Reply:
Yeah true, but with 14 with don't test neither (backwards compatibility | latest). We can pick up a strategy here and then have something in the e2e to cover the other one cc @pChondros

Reply:
True, since we are testing the production setup in e2e tests, we should be okay with picking the latest in these tests.